source/initial_temperature/adiabatic.cc

```diff
@@ -316,6 +394,10 @@ namespace aspect
 
       prm.enter_subsection ("Initial temperature model");
       {
+        Utilities::AsciiDataBase<dim>::declare_parameters(prm,
+                                                          "$ASPECT_SOURCE_DIR/data/initial-temperature/adiabatic/",
+                                                          "adiabatic.txt",
+                                                          "Adiabatic");
```

Comment:
I think this is why the tester does not work. This is the declare parameters function of the ascii data base class, but you're calling it in the parse_parameters function. Instead you need to move this up to line 315 and call the appropriate parse_parameters function here instead.